src/transformations/defmodel.hpp

```diff
@@ -0,0 +1,1890 @@
+    const std::string &horizontalOffsetUnit() const {
+        return mHorizontalOffsetUnit;
+    }
+
+    /** Unit of vertical offsets. Only "metre" is supported. */
+    const std::string &verticalOffsetUnit() const {
+        return mVerticalOffsetUnit;
+    }
+
+    /** Type of horizontal uncertainty. e.g "circular 95% confidence limit" */
+    const std::string &horizontalUncertaintyType() const {
+        return mHorizontalUncertaintyType;
+    }
+
+    /** Unit of horizontal uncertainty. Only "metre" is supported. */
+    const std::string &horizontalUncertaintyUnit() const {
+        return mHorizontalUncertaintyUnit;
+    }
+
+    /** Type of vertical uncertainty. e.g "circular 95% confidence limit" */
+    const std::string &verticalUncertaintyType() const {
+        return mVerticalUncertaintyType;
+    }
+
+    /** Unit of vertical uncertainty. Only "metre" is supported. */
+    const std::string &verticalUncertaintyUnit() const {
+        return mVerticalUncertaintyUnit;
+    }
+
+    /** Defines how the horizontal offsets are applied to geographic
+     * coordinates. Only "addition" and "geocentric" are supported */
+    const std::string &horizontalOffsetMethod() const {
+        return mHorizontalOffsetMethod;
+    }
+
+    /** Get the region within which the deformation model is defined.
+     * It cannot be calculated outside this region */
+    const SpatialExtent &extent() const { return mSpatialExtent; }
+
+    /** Defines the range of times for which the model is valid, specified
+     * by a first and a last value. The deformation model is undefined for
+     * dates outside this range. */
+    struct TimeExtent {
+        Epoch first{};
+        Epoch last{};
+    };
+
+    /** Get the range of times for which the model is valid. */
+    const TimeExtent &timeExtent() const { return mTimeExtent; }
+
+    /** Get an array of the components comprising the deformation model. */
+    const std::vector<Component> &components() const { return mComponents; }
+
+  private:
+    MasterFile() = default;
+
+    std::string mFileType{};
+    std::string mFormatVersion{};
+    std::string mName{};
+    std::string mVersion{};
+    std::string mLicense{};
+    std::string mDescription{};
+    std::string mPublicationDate{};
+    Authority mAuthority{};
+    std::vector<Link> mLinks{};
+    std::string mSourceCRS{};
+    std::string mTargetCRS{};
+    std::string mDefinitionCRS{};
+    std::string mReferenceEpoch{};
+    std::string mUncertaintyReferenceEpoch{};
+    std::string mHorizontalOffsetUnit{};
+    std::string mVerticalOffsetUnit{};
+    std::string mHorizontalUncertaintyType{};
+    std::string mHorizontalUncertaintyUnit{};
+    std::string mVerticalUncertaintyType{};
+    std::string mVerticalUncertaintyUnit{};
+    std::string mHorizontalOffsetMethod{};
+    SpatialExtent mSpatialExtent{};
+    TimeExtent mTimeExtent{};
+    std::vector<Component> mComponents{};
```

Comment:
You might want to add a ``whatever_else`` member (NL::json?), since this looks like a metadata struct, and those historically grow things as time goes on.

Reply:
hum, I'm not keen on doing that :-) The objective of this class is to have traditional C++ access to metadata. At worse people can access the extra info with NL::json itself